lib/werewolf/engine/state.rb

```diff
@@ -0,0 +1,46 @@
+module Werewolf
+  module Engine
+    class State
+      attr_reader :current, :turn_number, :day_number, :enumerator
+
+      STATES = [:dawn, :day, :dusk, :night]
+      TRANSITIONS = [:dawn, :dusk]
+
+      # We start at turn 0, :dawn on day 0
+      def initialize
+        @enumerator = STATES.cycle
+        @current = @enumerator.next
+        @index, @turn_number, @day_number = 0, 0, 0
+      end
+
+      # Advance to the next time period
+      def next
+        # increment the index and turn every time next is called
+        @index += 1
+        @turn_number += 1
+        # only increment the day if a full cycle has happened
+        @day_number += @index/STATES.length
```

Comment:
so this doesn't work cause turn number never rests to 0, so the value of `@turn_number/STATES.length` no longer becomes 0 or 1 after `@turn_number > STATES.length`